enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,549 @@
+  single-node developer deployments, multi-node production
+  deployments, and Internet-of-things devices, those use cases are not
+  addressed directly in this document.
+* This enhancement does not address reducing actual utilization. There
+  is no expectation that a cluster configured to use a small number of
+  cores for management services would offer exactly the same
+  performance as the default. It must be stable and continue to
+  operate reliably, but may respond more slowly.
+* This enhancement assumes that the configuration of management
+  services is done as part of installing the cluster, and cannot be
+  changed later. Future enhancements may address enabling or
+  reconfiguring the feature described here on an existing cluster.
+* This enhancement describes partitioning concepts that could be
+  expanded to be used for other purposes. Use cases for partitioning
+  workloads for other purposes may be addressed by future
+  enhancements.
+
+## Proposal
+
+[Previous attempts](https://github.com/openshift/enhancements/pull/628)
+to solve this problem focused on reducing or scaling requests so that
+the normal scheduling criteria could be used to safely place them on
+CPUs in the shared pool. This proposal reframes the problem, so that
+instead of considering "scaling" or "tuning" the requests we think about
+"isolating" or "partitioning" the workloads away from the non-management
+workloads. This view of the problem is more consistent with how the
+requirement was originally presented by the customer.
+
+We want to define "management workloads" in a flexible way. For the
+purposes of this document, "management workloads" include all
+OpenShift core components necessary to run the cluster, any add-on
+operators necessary to make up the "platform" as defined by telco
+customers, and operators or other components from third-party vendors
+that the customer deems as management rather than operational. It is
+important to note that not all of those components will be delivered
+as part of the OpenShift payload and some may be written by the
+customer or by vendors who are not our partners.
+
+The basic proposal is to provide a way to identify management
+workloads at runtime and to use CRI-O to run them on a user-selected
+set of CPUs, while other workloads will be prevented from running
+there. This effectively gives 3 pools of CPUs (shared, dedicated, and
+management) and means the shared CPU pool will need other CPUs to
+support burstable or best-effort workloads.
+
+We want the isolation to be in place from the initial buildout of the
+cluster, to ensure predictable behavior. Therefore, the feature must
+be enabled during installation. To enable the feature, the user will
+specify the set of CPUs to run all management workloads. When the
+management CPU set is not defined, the feature will be completely
+disabled.
+
+We generally want components to opt-in to being considered management
```

Comment:
why do we want to go with opt-in approach again? It's really had not to loose track of it, and requires high maintenance. Also, some yaml are templates and being rendered, others using knative, and more interesting approaches. Also, there are OLM operators - sriov, pao and others. Besides that, there are operatnds that are deployed only in certain cases (for example, registry in case of BM setup will not be deployed unless storage is configured). So it would be hard to get hermetic coverage in the CI.

Cannot we at least say the CVO does all the heavy lifting? If it recognizes a pod, it adds that annotation? It will solve 80% of the effort?